### PR TITLE
Import/export all settings into a zipped archive

### LINKED ISSUE #7032: Save favorites in the settings json
### Description

My files (2.0.36) has vertical tabs button off but when importing to files (2.0.39) it gets turned on again. Side bar pins and home screen widgets don't get saved or applied if they are saved.

### Steps To Reproduce

_No response_

### Expected behavior

It should correctly apply/save settings stated above

### Files Version

2.0.39.0

### Windows Version

10.0.19043.1348

### Relevant Assets

[debug.log](https://github.com/files-community/Files/files/7582928/debug.log)
[debug_fulltrust.log](https://github.com/files-community/Files/files/7582929/debug_fulltrust.log)

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #7032

**Details of Changes**
Add details of changes here.
- Extends the import/export command in About page to all settings. Settings are exported in a .zip file. It includes: pinned items, user settings, file tags, terminal profiles, bundles. On import, all settings are applied without restart.

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Comment by gave92** [comment]
@yair100 this is the proposal discussed in #7032

**Review by yair100 — DISMISSED** [review]

**Comment by gave92** [comment]
Uops pushed to the wrong branch, restored

**Review by yair100 — DISMISSED** [review]

File: Files/Filesystem/StorageItems/ZipStorageFile.cs [hunk 1]
```diff
@@ -391,20 +324,152 @@ public override IAsyncOperation<BaseStorageFolder> GetParentAsync()
...
+            }
+        }
+
+        public override string FolderRelativeId => $"0\\{Name}";
+
+        public override IStorageItemExtraProperties Properties => new BaseBasicStorageItemExtraProperties(this);
+
+        #region Private
+
+        private IAsyncOperation<ZipFile> OpenZipFileAsync(bool readWrite, bool openProtected = false)
```
Comment (d2dyno1) [inline Files/Filesystem/StorageItems/ZipStorageFile.cs:367]: Can you pass in `FileAccessMode`?

File: Files/Filesystem/StorageItems/ZipStorageFile.cs [hunk 2]
```diff
@@ -41,64 +34,33 @@ public override IAsyncOperation<StorageFile> ToStorageFileAsync()
...
-                                await outStream.FlushAsync();
-                            }
-                            request.Dispose();
-                        }
-                        else
-                        {
-                            request.FailAndClose(StreamedFileFailureMode.CurrentlyUnavailable);
+                            return null;
                         }
+                        return new FileStream(hFile, FileAccess.Read).AsRandomAccessStream();
```
Comment (d2dyno1) [inline Files/Filesystem/StorageItems/ZipStorageFile.cs:55]: Can this follow `accessMode` parameter?

**Review by d2dyno1 — CHANGES_REQUESTED** [review]

File: Common/FileTagsDb.cs [hunk 3]
```diff
@@ -149,6 +150,18 @@ public void Dispose()
             db.Dispose();
         }
 
+        public void Import(string json)
+        {
+            var dataValues = JsonSerializer.DeserializeArray(json);
```
[the remark this answers is not in the document]
Comment (gave92) [reply Common/FileTagsDb.cs:155]: Yeah but we have a catch in AboutViewModel. In any case I'll make this method coherent with the other ImportSettings methods.

File: Files/Controllers/SidebarPinnedController.cs [hunk 4]
```diff
@@ -26,51 +29,66 @@ public async Task InitializeAsync()
...
         {
-            Folder = ApplicationData.Current.LocalCacheFolder;
+            await LoadAsync();
+            Model.RemoveStaleSidebarItems();
+        }
 
-            try
+        private async Task LoadAsync()
+        {
+            StorageFolder Folder = await FilesystemTasks.Wrap(() => ApplicationData.Current.LocalFolder.CreateFolderAsync("settings", CreationCollisionOption.OpenIfExists).AsTask());
```
[the remark this answers is not in the document]
Comment (gave92) [reply Files/Controllers/SidebarPinnedController.cs:40]: No need, FilesystemResult is useful only if we care about why the call has failed. Here we really don't.

File: Files/Controllers/SidebarPinnedController.cs [hunk 5]
```diff
@@ -82,12 +100,40 @@ private async Task LoadAsync()
...
+                    serializer.Formatting = Formatting.Indented;
+                    serializer.Serialize(file, Model);
+                }
             }
+            catch
+            {
+            }
+        }
+
+        private async Task<IEnumerable<string>> ReadV1PinnedItemsFile()
```
[the remark this answers is not in the document]
Comment (gave92) [reply Files/Controllers/SidebarPinnedController.cs:117]: @yair100 should we drop support for old "PinnedItems.txt"? Don't quite remember how old is that.

File: Common/FileTagsDb.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (gave92) [reply Common/FileTagsDb.cs:155]: Actually most other ImportSettings() methods (e.g. UserSettings) don't do exception handling, keeping like this for now. The catch in AboutViewModel is enough.

File: Files/Controllers/SidebarPinnedController.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (gave92) [reply Files/Controllers/SidebarPinnedController.cs:117]: Not removed, but given priority to V2 which is what 99% of users will have.

**Review by d2dyno1 — APPROVED** [review]

**Review by yair100 — APPROVED** [review]

**Comment by yaichenbaum** [comment]
@gave92 thank you!

**Pull request merged**